modules/http-helper/http_helper.go

```diff
@@ -13,19 +14,23 @@ import (
 	"github.com/gruntwork-io/terratest/modules/retry"
 )
 
-// HttpGet performs an HTTP GET on the given URL and return the HTTP status code and body. If there's any error, fail the test.
-func HttpGet(t *testing.T, url string) (int, string) {
-	statusCode, body, err := HttpGetE(t, url)
+// HttpGet performs an HTTP GET, with an optional pointer to a custom TLS configuration, on the given URL and
+// return the HTTP status code and body. If there's any error, fail the test.
+func HttpGet(t *testing.T, url string, TLSConfig *tls.Config) (int, string) {
+	statusCode, body, err := HttpGetE(t, url, TLSConfig)
 	if err != nil {
 		t.Fatal(err)
 	}
 	return statusCode, body
 }
 
-// HttpGetE performs an HTTP GET on the given URL and return the HTTP status code, body, and any error.
-func HttpGetE(t *testing.T, url string) (int, string, error) {
+// HttpGetE performs an HTTP GET, with an optional pointer to a custom TLS configuration, on the given URL and
+// return the HTTP status code, body, and any error.
+func HttpGetE(t *testing.T, url string, TLSConfig *tls.Config) (int, string, error) {
 	logger.Logf(t, "Making an HTTP GET call to URL %s", url)
 
+	http.DefaultTransport.(*http.Transport).TLSClientConfig = TLSConfig
```

Comment:
I'm not sure this is a good idea, as this updates the http library globally. This can have negative side effects for other libraries that depend on the http library.

It would be better to only set it on the client object being created in the context of this function. You can do that by wrapping the config on the transport:

```
tr := &http.Transport{
    TLSClientConfig: tlsConfig,
}

c := &http.Client{Transport: tr}
```

Reply:
Good shout.  Will update.